packages/did/src/DidDetails/FullDidDetails.ts

```diff
@@ -36,168 +34,146 @@ import {
-    }: {
-      keySelection?: DidKeySelectionCallback<DidVerificationKey>
-      txCounter?: BN
-    } = {}
-  ): Promise<SubmittableExtrinsic> {
-    const signingKey = await keySelection(this.getKeysForExtrinsic(extrinsic))
-    if (!signingKey) {
-      throw new SDKErrors.DidError(
-        `The details for DID "${this.uri}" do not contain the required keys for this operation`
-      )
-    }
-    return generateDidAuthenticatedTx({
-      didIdentifier: this.identifier,
-      signingPublicKey: signingKey.publicKey,
-      alg: getSigningAlgorithmForVerificationKeyType(signingKey.type),
-      sign,
-      call: extrinsic,
-      txCounter: txCounter || (await this.getNextNonce()),
-      submitter: submitterAccount,
-    })
+  return details
+}
+
+/**
+ * Returns all the DID keys that could be used to sign the provided extrinsic for submission.
+ * This function should never be used directly by SDK users, who should rather call [[FullDidDetails.authorizeExtrinsic]].
+ *
+ * @param did The DID data.
+ * @param extrinsic The unsigned extrinsic to perform the lookup.
+ *
+ * @returns All the keys under the full DID that could be used to generate valid signatures to submit the provided extrinsic.
+ */
+export function getKeysForExtrinsic(
+  did: DidDetails,
+  extrinsic: Extrinsic
+): DidVerificationKey[] {
+  const keyRelationship = getKeyRelationshipForExtrinsic(extrinsic)
+  return keyRelationship === 'paymentAccount' ? [] : did[keyRelationship] || []
+}
+
+/**
+ * Returns the next nonce to use to sign a DID operation.
+ * Normally, this function should not be called directly by SDK users. Nevertheless, in advanced cases where there might be race conditions, this function can be used as the basis on which to build parallel operation queues.
+ *
+ * @param did The DID data.
+ * @returns The next valid nonce, i.e., the nonce currently stored on the blockchain + 1, wrapping around the max value when reached.
+ */
+export async function getNextNonce(did: DidDetails): Promise<BN> {
+  const currentNonce = await queryNonce(did.identifier)
+  return increaseNonce(currentNonce)
+}
+
+/**
+ * Signs and returns the provided unsigned extrinsic with the right DID key, if present. Otherwise, it will throw an error.
+ *
+ * @param did The DID data.
+ * @param extrinsic The unsigned extrinsic to sign.
+ * @param sign The callback to sign the operation.
+ * @param submitterAccount The KILT account to bind the DID operation to (to avoid MitM and replay attacks).
+ * @param signingOptions The signing options.
+ * @param signingOptions.keySelection The optional key selection logic, to choose the key among the set of allowed keys. By default it takes the first key from the set of valid keys.
+ * @param signingOptions.txCounter The optional DID nonce to include in the operation signatures. By default, it uses the next value of the nonce stored on chain.
+ * @returns The DID-signed submittable extrinsic.
+ */
+export async function authorizeExtrinsic(
```

Comment:
So now this function takes a `DidDetails` instance, right? It means that it is not possible to prevent the `authorizeExtrinsic` to be used with light DIDs?

Reply:
Maybe we can check the DID uri, to distinguish between Light and Full Dids and throw an early error here.

Reply:
As discussed f2f, this would be a good compromise, with the goal that stuff should fail before it makes its way to the full node.

Reply:
Good point, it will also make more explicit why we even have the distinction

Reply:
I’ve pushed a fix for that in d3f19b18